common/static/common/js/discussion/discussion_router.js

```diff
@@ -89,16 +100,44 @@
                 if (this.thread) {
                     return this.nav.setActiveThread(this.thread.get("id"));
                 } else {
-                    return this.nav.goHome;
+                    return this.nav.goHome();
                 }
             };
 
             DiscussionRouter.prototype.showThread = function(forum_name, thread_id) {
+                var self;
+                self = this;
                 this.thread = this.discussion.get(thread_id);
+
+                if (this.thread) {
+                    this.renderThreadView();
+                    return;
+                }
+
+                // if thread is not loaded yet for some reason - try loading it
+                DiscussionUtil.safeAjax({
+                    url: DiscussionUtil.urlFor('retrieve_single_thread', forum_name, thread_id)
+                }).done(function(data) {
+                    // if succeded - proceed normally
+                    self.thread = new Thread(data.content);
+                    self.discussion.add(self.thread);
+                    self.renderThreadView();
+                }).fail(function(xhr) {
+                    // otherwise display error message and navigate to all threads view
+                    var errorMessage = (xhr.status === 404) ?
```

Comment:
Multiline ternaries are difficult to parse (for me at least), suggest just using a conventional `if` here.